quinn-proto/src/connection/mod.rs

```diff
@@ -3095,15 +3095,20 @@ impl Connection {
         }
 
         // DATAGRAM
+        let mut sent_datagrams = false;
         while buf.len() + Datagram::SIZE_BOUND < max_size && space_id == SpaceId::Data {
             match self.datagrams.write(buf, max_size) {
                 true => {
+                    sent_datagrams = true;
                     sent.non_retransmits = true;
                     self.stats.frame_tx.datagram += 1;
                 }
                 false => break,
             }
         }
+        if sent_datagrams {
+            self.events.push_back(Event::DatagramSent);
```

Comment:
I'm a bit concerned at how spammy this might be. Would it make sense to emit an event only once after each time we return `Blocked`?